### PR TITLE
feat(Icon): Add disabled and disabledStyle

### PR DESCRIPTION
**Pull request by iRoachie**
Adds `disabled` and `disabledStyle` props to the Icon component. 

Ideally, we should do this for all our interactable components. It's a common use case for like forms, where when submitting the fields are disabled.

Usage:

```jsx
<Icon disabled />
<Icon disabled disabledStyle={{ backgroundColor: 'pink' }} />
```

### CODE REVIEW

File: src/icons/Icon.js [hunk 1]
```diff
@@ -46,11 +48,17 @@ const Icon = props => {
             alignItems: 'center',
             justifyContent: 'center',
           },
-        ]}
+          disabled && styles.disabled,
+          disabled && disabledStyle,
+        ])}
+        {...onPress && { disabled }}
```
Comment (Gregjarvez) [inline src/icons/Icon.js]: just to improve readability should should we put the conditional statement in parens ? ` {...(onPress && { disabled })}`

Comment (iRoachie) [reply src/icons/Icon.js]: Its an object so we have to spread it. What this is doing is only adding disabled to the props if onPress is passed in.

Comment (Gregjarvez) [reply src/icons/Icon.js]: yeah we do spread. I am just saying we wrap that expression in parens to enhance readability 😉

Comment (iRoachie) [reply src/icons/Icon.js]: Oh lol, prettier removes unnecessary brackets.

**Pull request merged**